### PR TITLE
[FEAT]: Improvements to noise measurement in local mags sub-module

### PR DESCRIPTION
**Pull request by TomWinder**
## What is the purpose of this Pull Request? 
The main goal of this PR is to add new options for measuring and using noise amplitudes when calculating local magnitudes. Highlights are:
 * A new noise measure: the mean amplitude of the signal envelope https://github.com/QuakeMigrate/QuakeMigrate/pull/117/commits/d88f3ca7a5b5a64484c80e1ccf96b6099e991103
 * Measure the average amplitude in the signal windows with the same method used for the noise window https://github.com/QuakeMigrate/QuakeMigrate/pull/117/commits/a4b868a350f9f694cb28d3dec5afe27ac43df5ff
 * Add a new (experimental) option to further improve the power of the mag_r_squared statistic to distinguish between real events and artefacts. https://github.com/QuakeMigrate/QuakeMigrate/pull/117/commits/615a68edb22cd93e6009dfbee247839ae309164d

## Test cases
Tested on the examples (particularly the dike example), as well as the Askja regional dataset.

- [x] All examples run without any new warnings
- [x] test_benchmarks.py reports all example tests pass -- No! Changes to .amps file.

### System details
Ubuntu 18.04

### Final checklist
- [x] `develop` base branch selected?
- [x] Update benchmarks.
- [x] All tests still pass.
- [x] Any new or changed features are fully documented.

### CODE REVIEW

**Review by hemmelig — COMMENTED** [review]
Again, all looks good and I will test tomorrow on the Borneo dataset. From my reading, it looks like I have to play with:

1. Different averaging methods
2. The experimental filtering method. Can you give me some details on that?
3. Anything else?

**Comment by TomWinder** [comment]
> Again, all looks good and I will test tomorrow on the Borneo dataset. From my reading, it looks like I have to play with:
> 
> 1. Different averaging methods
> 2. The experimental filtering method. Can you give me some details on that?
> 3. Anything else?

1: Yep: just added the envelope method and moved to a function, so no major changes. Using the same option (e.g. "RMS") should give the same results as previously. I don't foresee any problems with "ENV".

2: Pretty simple - see the commit message for a full description: https://github.com/QuakeMigrate/QuakeMigrate/pull/117/commits/615a68edb22cd93e6009dfbee247839ae309164d (have also added these to the PR now - mb.)

3: I didn't actually check that the updated `to_obspy` still works on old .amps files, though it should. Works fine for me on the new ones.

**Comment by hemmelig** [comment]
Checked all of the above. One minor issue related to an untested toggle - if the `bandpass_filter` option in the amplitude parameters was toggled _off_, an error would occur due to an un-assigned variable (`filter_gain`).

Error message:
```
Traceback (most recent call last):
  File "locate.py", line 116, in <module>
    scan.locate(starttime=start_time, endtime=end_time)
  File "/raid1/cb793/miniconda3/envs/QuakeMigrate/lib/python3.7/site-packages/quakemigrate/signal/scan.py", line 331, in locate
    self._locate_events(starttime=starttime, endtime=endtime)
  File "/raid1/cb793/miniconda3/envs/QuakeMigrate/lib/python3.7/site-packages/quakemigrate/signal/scan.py", line 453, in _locate_events
    event, _ = self.mags.calc_magnitude(event, self.lut, self.run)
  File "/raid1/cb793/miniconda3/envs/QuakeMigrate/lib/python3.7/site-packages/quakemigrate/util.py", line 610, in wrapper
    result = func(*args, **kwargs)
  File "/raid1/cb793/miniconda3/envs/QuakeMigrate/lib/python3.7/site-packages/quakemigrate/signal/local_mag/local_mag.py", line 189, in calc_magnitude
    amps = self.amp.get_amplitudes(event, lut)
  File "/raid1/cb793/miniconda3/envs/QuakeMigrate/lib/python3.7/site-packages/quakemigrate/util.py", line 610, in wrapper
    result = func(*args, **kwargs)
  File "/raid1/cb793/miniconda3/envs/QuakeMigrate/lib/python3.7/site-packages/quakemigrate/signal/local_mag/amplitude.py", line 350, in get_amplitudes
    filter_sos)
  File "/raid1/cb793/miniconda3/envs/QuakeMigrate/lib/python3.7/site-packages/quakemigrate/signal/local_mag/amplitude.py", line 791, in _measure_signal_amps
    np.abs(filter_gain[0]))
UnboundLocalError: local variable 'filter_gain' referenced before assignment
```

Once this has been addressed, I'm happy to approve this PR and merge (pending benchmark updates).

**Review by hemmelig — APPROVED** [review]

**Pull request merged**
